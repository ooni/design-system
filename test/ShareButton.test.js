/* global describe, test, expect */
import React from 'react'
import { renderWithTheme } from './index'
import { TwitterShareButton, FacebookShareButton } from '../'

describe('TwitterShareButton', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <TwitterShareButton />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('FacebookShareButton', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <FacebookShareButton />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
