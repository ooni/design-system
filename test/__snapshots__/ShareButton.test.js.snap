// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookShareButton renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #0f77b8;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-filter: brightness(125%);
  filter: brightness(125%);
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: none;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #f8f9fa;
  background-color: #3b5998;
  border-radius: 20px;
  border-width: 2px;
  font-size: 16px;
}

.c1:hover:enabled {
  background-color: #37a6ed;
}

.c1:active:enabled {
  background-color: #0f77b8;
}

.c1:enabled {
  cursor: pointer;
}

.c1:hover {
  -webkit-filter: brightness(90%);
  filter: brightness(90%);
}

.c1:active {
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
  -webkit-transition: 0.2s ease-in;
  transition: 0.2s ease-in;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 8px;
}

<a
  class="c0"
  href="https://www.facebook.com/sharer/sharer.php?&u=undefined"
>
  <button
    class="c1"
    font-size="1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
      />
    </svg>
    <span
      class="c2"
    >
      Share
    </span>
  </button>
</a>
`;

exports[`TwitterShareButton renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #0f77b8;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-filter: brightness(125%);
  filter: brightness(125%);
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: none;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #f8f9fa;
  background-color: #1da1f2;
  border-radius: 20px;
  border-width: 2px;
  font-size: 16px;
}

.c1:hover:enabled {
  background-color: #37a6ed;
}

.c1:active:enabled {
  background-color: #0f77b8;
}

.c1:enabled {
  cursor: pointer;
}

.c1:hover {
  -webkit-filter: brightness(90%);
  filter: brightness(90%);
}

.c1:active {
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
  -webkit-transition: 0.2s ease-in;
  transition: 0.2s ease-in;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 8px;
}

<a
  class="c0"
  href="https://twitter.com/intent/tweet?text=undefined&url=undefined&hashtags=undefined&hashtags=undefined&in-reply-to=undefined"
>
  <button
    class="c1"
    font-size="1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
      />
    </svg>
    <span
      class="c2"
    >
      Tweet
    </span>
  </button>
</a>
`;
